src/types.ts

```diff
@@ -28,12 +28,14 @@ export type GenerateAuthUrlOptions =
   | {
       code_challenge_method?: string;
       code_challenge?: string;
+      testnet: boolean;
```

Comment:
I think we should pass an `authorizeUrl?: string` here for better flexibility, and make it optional.

Also can you update the type so it is `{authorizeUrl?: string} &( A | B | C)` so you don't need to duplicate the declaration

Reply:
in that case 
1. this also requires validation. if client is using right url or not

Reply:
user has only two optoins. default mainnet. if they wish to setup testnet they just set a boolean. isn't this more convenient?

Reply:
We could also point it at a local running instance for example. @bumi do you have any preference here?

Reply:
resolved

Reply:
+1 for setting the URL

Reply:
made suggested changes